### ISSUE TITLE
.fit() fails with 1 segment

### ISSUE THREAD

**Issue by pabloslash** [cjekel/piecewise_linear_fit_py#100]
Is there a reason why the fit cannot be done for a single (1-segment) linear regression?

**Comment by danhamill** [comment]
I came across this the other day and am also interested in the answer.

**Comment by cjekel** [comment]
So the expected api would be able to do something like `.fit(1)`? 

1 segment would be a special routine. You don't need the underlying optimization routine to move 'breakpoints' around.

I guess all the underlying mechanisms are there to support fits of 1 segment, I would just need to write this special routine, then query it when `.fit()` or `.fitfast()` were called.

**Comment by cjekel** [comment]
that routine would be

```python
def fit_with_one_segment(self):
    self.fit_with_breaks([self.break_0, self.break_n])
```

**Comment by cjekel** [comment]
Thanks for the suggestion! I've added it to the code, and released a new version 2.2.1.

You can now do `fit` and `fitfast` for one line segment. In both cases, simple linear regression is applied.
